torch/csrc/jit/codegen/cuda/lower_scalar_hoist.cpp

```diff
@@ -12,10 +12,28 @@ namespace cuda {
 
 namespace {
 
+// Get the position of the innermost non-trivial loop
+int64_t getInnermostNonTrivialLoop(const std::vector<kir::ForLoop*>& loops) {
+  int64_t position = -1;
+  for (auto i : c10::irange(loops.size())) {
+    if (!loops.at(i)->isTrivial()) {
+      position = i;
+    }
+  }
+  return position;
+}
+
 // Find the outer-most loop nest that contains all the dependencies of `value`.
 int64_t findOutermostPosWithSatisfiedDependency(
     Val* value,
     const std::vector<kir::ForLoop*>& loops) {
+  // We don't recursively look into tensor indexing to find its dependency.
```

Comment:
Is this a compromise? IIUC, this could happen with indirect indexing such as the index tensor of an `index_select`. The dependency of the index tensor should be at the computeAt position of the tensor, so technically the position can be outer than the innermost non-trivial loop, right?

Don't think it would matter in practice, but just want to clarify.